unmarshal_test.go

```diff
@@ -2316,19 +2317,110 @@ type handleJSONDataChild struct {
 	Data map[string]interface{} `json:"-"`
 }
 
-func (c *handleJSONDataChild) HandleJSONData(data map[string]interface{}) {
+func (c *handleJSONDataChild) HandleJSONData(data map[string]interface{}) error {
+	if _, exists := data["fail"]; exists {
```

Comment:
1. Do we want to handle a scenario when `fail` exists but the value is false? 
2. Although it is used once,  maybe we should move `fail` to a constant instead of a hard-coded string.